src/exceptions.rs

```diff
@@ -340,6 +403,72 @@ impl StopIteration {
     }
 }
 
+impl std::fmt::Debug for BaseException {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        // Sneaky: we don’t really need a GIL lock here as nothing should be able to just mutate
+        // the "type" of an object, right? RIGHT???
+        //
+        // let gil = Python::acquire_gil();
+        // let _py = gil.python();
+        let py_type_name = unsafe { CStr::from_ptr((*(*self.as_ptr()).ob_type).tp_name) };
+        let type_name = py_type_name.to_string_lossy();
+        f.debug_struct(&*type_name)
+            // TODO: print out actual fields!
```

Comment:
We can access actual field only by type name?
Or we need to get the global `PyExc_~`s here?

Reply:
By "actual fields" I meant the fields of the `PyException_HEAD` structure component, which is defined as such:

```
#define PyException_HEAD PyObject_HEAD PyObject *dict;\
             PyObject *args; PyObject *traceback;\
             PyObject *context; PyObject *cause;\
             char suppress_context;
```

At very least values of `args`, `traceback`, `context`, `cause` and `suppress_context` are interesting, but I can imagine it being useful to also print out at least the `refcount` from `PyObject_HEAD` too.

`f.debug_struct(&*type_name)` is just a builder for debug output. See https://doc.rust-lang.org/stable/std/fmt/struct.Formatter.html#method.debug_struct. The final result should look something like

```
f.debug_struct(type_name).field("args", &self.args).field("traceback", &self.traceback)/* etc... */.finish()
```